### PR TITLE
ci: Upgrade frontend dev tooling (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Summary
This PR updates 
- vue
- vue-router
- pinia
- vite-plugin-dts
- vue-tsc
- @sentry/vite-plugin
- storybook
- esbuild

## Review / Merge checklist

- [x] PR title and summary are descriptive

### CODE REVIEW

File: cypress/package.json [hunk 1]
```diff
@@ -15,17 +15,17 @@
   },
   "devDependencies": {
```
Comment (mutdmour) [inline cypress/package.json:16]: would be nice if we can automate checking for this. For example, on each build, we get a log with all duplicated dependencies.

File: packages/editor-ui/src/views/SettingsUsersView.test.ts [hunk 2]
```diff
@@ -119,6 +119,7 @@ describe('SettingsUsersView', () => {
 	it('disables the invite button on SAML login', async () => {
 		const pinia = createTestingPinia({ initialState: getInitialState() });
 		const ssoStore = useSSOStore(pinia);
+		// @ts-expect-error updating a readonly property
 		ssoStore.isSamlLoginEnabled = true;
```
Comment (mutdmour) [inline packages/editor-ui/src/views/SettingsUsersView.test.ts:123]: not pretty. would be nice if we enabled this by setting the settings here instead.

File: packages/editor-ui/src/components/NodeCredentials.vue [hunk 3]
```diff
@@ -510,7 +510,7 @@ function getCredentialsFieldLabel(credentialType: INodeCredentialDescription): s
 					data-test-id="node-credentials-select"
 				>
 					<N8nSelect
-						:model-value="getSelectedId(credentialTypeDescription.name)"
+						:model-value="getSelectedId(credentialTypeDescription.name)!"
```
Comment (mutdmour) [inline packages/editor-ui/src/components/NodeCredentials.vue:513]: Can we just update the type for modalValue to accept `undefined`?

File: packages/design-system/src/components/N8nFormInput/FormInput.vue [hunk 4]
```diff
@@ -219,7 +219,7 @@ defineExpose({ inputRef });
 				v-else-if="type === 'select' || type === 'multi-select'"
 				ref="inputRef"
 				:class="{ [$style.multiSelectSmallTags]: tagSize === 'small' }"
-				:model-value="modelValue"
+				:model-value="modelValue!"
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nFormInput/FormInput.vue:222]: same question here. Could we update prop types for modal value to fix this?

**Review by mutdmour — COMMENTED** [review]
Thanks for this. Just a few comments

File: packages/editor-ui/src/views/SettingsUsersView.test.ts (hunk 2 above)
Comment (netroy) [reply packages/editor-ui/src/views/SettingsUsersView.test.ts:123]: there is more to refactor here than what's in the scope of this PR. maybe another time.

**Review by mutdmour — DISMISSED** [review]
thanks 👏🏽

File: packages/design-system/src/components/N8nSelect/Select.vue [hunk 5]
```diff
@@ -1,56 +1,33 @@
...
+		popperClass?: string;
+		popperAppendToBody?: boolean;
+		limitPopperWidth?: boolean;
+		noDataText?: string;
+	}>(),
+	{
+		size: 'large',
+		teleported: true,
 	},
-	placeholder: {
```
Comment (MiloradFilipovic) [inline packages/design-system/src/components/N8nSelect/Select.vue:17]: Finally! Thanks for taking care of these!

**Review by MiloradFilipovic — DISMISSED** [review]
Looks good on my end. Thanks for this!

**Review by MiloradFilipovic — COMMENTED** [review]
Code looks good but looks like some [e2e tests are still failing](https://cloud.cypress.io/projects/5hbsdn/runs/6932/overview?roarHideRunsWithDiffGroupsAndTags=1).
When running e2e locally, I am also having `42-nps-survey.cy.ts` failing (alongside `17-sharing.cy.ts` and `39-projects.cy.ts`)

**Review by MiloradFilipovic — APPROVED** [review]
Looks good to me, builds and seems to work well locally.
I [also ran the e2e tests](https://github.com/n8n-io/n8n/actions/runs/11270208837) but then saw that you already [ran those yesterday](https://github.com/n8n-io/n8n/actions/runs/11257945317) so good to go on my end 🙌🏻

**Comment by janober** [comment]
Got released with [`n8n@1.64.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.64.0)

**Pull request merged**